DROOLS-6943 DMN doc update built-in function DMNv1.4 (#4363)
kiegroup/drools: kie-dmn/ref-dmn-feel-builtin-functions.adoc

----
--
// ----------------------------------------------------------------------------

context merge( contexts )::
+
--
Returns a new context that includes all entries from the given contexts; if some of the keys are equal, the entries are overriden.
The entries are overridden in the same order as specified by the supplied parameter, with new entries added as the last entry in the new context.

.Parameters
[cols="30%,70%", options="header"]
|===
|Parameter
|Type

|`contexts`
|`list` of `context`
|===

.Examples
[source,FEEL]
----
context merge([{x:1}, {y:2}]) = {x:1, y:2}
context merge([{x:1, y:0}, {y:2}]) = {x:1, y:2}
----
--
// ----------------------------------------------------------------------------

== Miscellaneous functions

These functions provide support utilities for several miscellaneous use-cases.
For example, when a decision depends on the current date, like deciding the support SLA over the weekends, additional charges for weekend delivery, etc.

It is important to note that the functions in this section are intended to be side-effect-free, but they are not deterministic and not idempotent from the perspective of an external observer.

> As a user you are encouraged to isolate the decision logic that uses these functions in specific DRG elements, such as Decisions.

Reviewer: ```suggestion
As a user, you are encouraged to isolate the decision logic that uses these functions in specific DRG elements, such as Decisions.
```

The encapsulation enables them to be overridden with synthetic values during Scenario Testing, that remain constant across executions of the DMN model's test cases.

now()::
+
--
Returns the current `date and time`.

.Examples
[source]
----
now()
----
--
// ----------------------------------------------------------------------------

today()::
+
--
Returns the current `date`.

.Examples
[source]
----
today()
----
--
// ----------------------------------------------------------------------------

== KIE Extended functions

These functions are provided as extension to the DMN Standard to enable support for several miscellaneous use-cases on top of the Drools DMN Engine.

invoke( namespace, modelName, decisionName, parameters )::
+
--

> Returns the result of the decision evaluation in the specified DMN model available to the `DMNRuntime` environment the current DMN model is executed in.

Reviewer: ```suggestion
Returns the result of the decision evaluation in the specified DMN model available to the `DMNRuntime` environment in the current DMN model is executed.
```

> This function is _deprecated_ in favour of encouraging the usage of DMN Standard capabilities wherever possible; since DMNv1.2 it shall be possible to use the DMN standard's _Import_ functionality to import Business Knowledge Model (BKM) nodes and Decision Service nodes, in order to be invoked from another model.

Reviewer: ```suggestion
This function is _deprecated_ in favor of encouraging the usage of DMN Standard capabilities wherever possible; since DMNv1.2 it shall be possible to use the DMN standard's _Import_ functionality to import Business Knowledge Model (BKM) nodes and Decision Service nodes, to be invoked from another model.
```
